src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/action/explain/TransportExplainRollupAction.kt

```diff
@@ -69,60 +71,73 @@ class TransportExplainRollupAction @Inject constructor(
                     }
                 }
             ))
-        client.search(searchRequest, object : ActionListener<SearchResponse> {
-            override fun onResponse(response: SearchResponse) {
-                try {
-                    response.hits.hits.forEach {
-                        val rollup = contentParser(it.sourceRef).parseWithType(it.id, it.seqNo, it.primaryTerm, Rollup.Companion::parse)
-                        idsToExplain[rollup.id] = ExplainRollup(metadataID = rollup.metadataID)
+        client.threadPool().threadContext.stashContext().use {
+            client.search(searchRequest, object : ActionListener<SearchResponse> {
+                override fun onResponse(response: SearchResponse) {
+                    try {
+                        response.hits.hits.forEach {
+                            val rollup = contentParser(it.sourceRef).parseWithType(
+                                it.id,
+                                it.seqNo,
+                                it.primaryTerm,
+                                Rollup.Companion::parse
+                            )
+                            idsToExplain[rollup.id] = ExplainRollup(metadataID = rollup.metadataID)
+                            rolesMap[rollup.id] = rollup.user?.roles
```

Comment:
Is there any reason to not show the default roles - cause that is what we are setting internally in those cases right?

Reply:
If user is null, it prob is a job from previous version. We may not need to show "all_access" roles for it here, but just call out the old job will run as before (run by "all_access" role)